### PR TITLE
add documentation

### LINKED ISSUE #7: scaladoc
Scaladoc for all traits, classes, and methods

### LINKED ISSUE #8: A microsite
https://zio.github.io/zio-keeper/

### PR DESCRIPTION
**Pull request by regis-leray**
first draft of documentation.

- [x] javadoc
- [x] microsite

closes #7 
closes #8

### CODE REVIEW

**Comment by jdegoes** [comment]
Out of sync with base branch, then good to merge!

**Comment by regis-leray** [comment]
You went too fast, im still missing a microsite ;)
I will ping tou again when it will be ready

File: .circleci/config.yml [hunk 1]
```diff
@@ -83,7 +91,73 @@ release: &release
...
+        [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+        nvm install
+        nvm use
+        node -v
+
+microsite: &microsite
+  steps:
+    - add_ssh_keys:
+        fingerprints:
+          - "11:2c:fa:98:69:d8:a1:d5:76:2e:38:d0:91:39:85:ce"
```
Comment (softinio) [inline .circleci/config.yml]: We need to change this fingerprint

**Comment by softinio** [comment]
@regis-leray Do you want to add the about section of the `zio` microsite to this repo's microsite also:

https://github.com/zio/zio/tree/master/docs/about

File: .circleci/config.yml (hunk 1 above)
Comment (regis-leray) [reply .circleci/config.yml]: which one shoud i use ?

Comment (softinio) [reply .circleci/config.yml]: I need to create a new one for this repo will get it done and either message you with it or do a PR to this branch/PR for you

created this to track/remind myself: https://github.com/zio/zio-ftp/issues/22

Comment (softinio) [reply .circleci/config.yml]: @regis-leray done. Can you change it in this PR to: 

```
68:95:b1:fc:a3:eb:4e:85:45:74:07:d3:c0:35:c8:ec
```

Thanks.

Comment (regis-leray) [reply .circleci/config.yml]: thanks i did it

**Comment by regis-leray** [comment]
@jdegoes i created a simple microsite, and add basic javadoc for now

**Comment by jdegoes** [comment]
@regis-leray Superb! Just got one test failing looks like.

**Comment by regis-leray** [comment]
@jdegoes everything shoul be good now. It was circeci failure.

**Pull request merged**